pygfx/objects/_skins.py

```diff
@@ -0,0 +1,206 @@
+import numpy as np
+import pylinalg as la
+from typing import List
+from ._base import WorldObject
+from ..utils import array_from_shadertype
+from ..utils.transform import AffineBase, callback
+from ..utils.enums import BindMode
+from ..resources import Buffer
+from ._more import Mesh
+
+
+class Bone(WorldObject):
+    """A bone.
+
+    A bone is a part of a skeleton.The skeleton in turn is used by the SkinnedMesh.
+    Bones are almost identical to a blank WorldObject.
+
+    """
+
+    def __init__(self, name=""):
+        super().__init__(name=name)
+        self.visible = False
+
+
+class Skeleton:
+    """A skeleton.
+
+    A skeleton is a collection of bones that are used to animate a SkinnedMesh.
+
+    Parameters
+    ----------
+    bones : List[Bone]
+        The array of bones.
+        Note this is a copy of the original array, not a reference, so you can modify the original array without effecting this one
+    bone_inverses : List[np.ndarray] | None
+        An array of matrix4x4 that represent the inverse of the world matrix of the individual bones.
+        If not provided, they will be auto calculated.
```

Comment:
Avoid confusion with ndarray:
```suggestion
    bones : List[Bone]
        The list of bones.
        Note that a copy will be made, so you can modify the original list without effecting this one.
    bone_inverses : List[np.ndarray] | None
        A list of matrix4x4 that represent the inverse of the world matrix of the individual bones.
        If not provided, they will be auto calculated.
```